examples/shell/shell_common/cmd_ping.cpp

```diff
@@ -143,19 +143,12 @@ CHIP_ERROR SendEchoRequest(streamer_t * stream)
 
     Messaging::SendFlags sendFlags;
     System::PacketBufferHandle payloadBuf;
-    char * requestData = nullptr;
 
-    uint32_t size = gPingArguments.GetEchoReqSize();
-    VerifyOrExit(size <= kMaxPayloadSize, err = CHIP_ERROR_MESSAGE_TOO_LONG);
-
-    requestData = static_cast<char *>(chip::Platform::MemoryAlloc(size));
-    VerifyOrExit(requestData != nullptr, err = CHIP_ERROR_NO_MEMORY);
-
-    snprintf(requestData, size, "Echo Message %" PRIu64 "\n", gPingArguments.GetEchoCount());
-
-    payloadBuf = MessagePacketBuffer::NewWithData(requestData, size);
+    payloadBuf = MessagePacketBuffer::New(gPingArguments.GetEchoReqSize());
     VerifyOrExit(!payloadBuf.IsNull(), err = CHIP_ERROR_NO_MEMORY);
 
+    payloadBuf->SetDataLength(gPingArguments.GetEchoReqSize());
```

Comment:
Please memset the part of the buffer up to the new data length to 0 or otherwise initialize it.  Otherwise we are possibly leaking data that was in previous messages.  Might be OK here, but in general it's a really bad pattern and we should not have it.

Reply:
Acked